### ISSUE TITLE
Sent live videos >10s freezing + 'Show More Info on Profile' no longer working

### ISSUE THREAD

**Issue by Magna-Graecia** [rodit/SnapMod#180]
Just updated to the latest Snapchat + SnapMod and was testing features as I usually do to see if anything is broken. Sending videos >10s with 'Disable Snap Splitting' turned off causes them to freeze on the recipient's end. The first 10-second clip plays fine, then the 2nd one will freeze midway, and what I noticed is that the timer bubble in the upper right-hand corner keeps going, but it slows down. When it empties, it will move to the next 10-second clip, and the timer bubble emptying will now be even slower. It will empty progressively slower as you move to the next clip. All clips after the first one will freeze midway. Tested with a 57-second long video that I've been using to test SnapMod since I began using it, so nothing has changed in that regard.

Enabling 'Disable Snap Splitting' eliminates any freezing issues. It only happens when snaps are being split.

And as the title states, 'Show More Info on Profile' no longer works. It shows the default 'Friends with blahblah since so and so" and no other info.

Can anyone else test and see if they experience the same to eliminate the possibility that it's a local issue?

@rodit P.S.- As a sidenote, I discovered a little tidbit during my testing. If you recall, I had remarked on the whole 'slingshotting' effect that happens when viewing a chat with a newly received message on stealth mode. When viewing the chat, it 'slingshots' you to the bottom to the newly received chats unless you hold your finger on the screen to stop it. What I discovered is that if I receive a chat, view the chat with stealth mode enabled, press and hold on the latest message to get the context menu, then leave the chat and kill Snapchat and restart it, I can now go back into the same chat with stealth mode enabled and freely scroll up with no slingshotting effect. If I leave the chat and go back into it, the slingshotting effect again returns. Strange, but at least it's a way to scroll up in the conversation with stealth mode enabled if need be haha.

**Comment by tarek133** [comment]
Can confirm the "show more info on profile" is no longer working.
Can't say anything abt the freezing issue for now though, as I can't test it

**Comment by Magna-Graecia** [comment]
@tarek133 Thanks for trying, and also for confirming the 'Show More Info on Profile' issue. Which way did you send the 1min+ video? If you opened the gallery drawer in a chat and simply tapped on the video (checkmarked it) and sent, then it will get sent with splits disabled no matter if 'Disable Snap Splitting' is enabled or not. You have to send by either going into gallery drawer in a chat and pressing and holding on a video there to load it into the camera edit window, then sending. Or you need to share the video from gallery to Snapchat and then send.

**Comment by rodit** [comment]
Thanks again for the testing - I'll have a look. The more profile info is an easy fix but the other issue seems rather unusual. I cannot recreate the freezing videos - they load and play fine on the recipients end after being split on my devices.

**Comment by Manjit-Gill** [comment]
Other than freezing on snap splitting . I have both the issues. The slingshot issue and no more information on profile. The more information issue started with latest update

**Comment by rodit** [comment]
The slingshotting probably isn't something I will fix unless I come across the solution by chance.

**Comment by Magna-Graecia** [comment]
Maybe the freezing issue is localized to me and people with my same OS version. Just tried sending a different video for verification (which I also had sent in the past), and it happens with that video as well, so it's not the video that's the issue. I guess I'll have to downgrade to the previous version of Snapchat and SnapMod for the time being.

**Comment by Magna-Graecia** [comment]
@rodit So I downgraded to 1.7.2 and 11.79.0.34 (84609) and hilariously enough, I still have the problem. Wtf? Now, the device I'm sending to (unrooted/unmodded) is running the latest Snapchat, so maybe it has something to do with that and not SnapMod?

On the plus side, 'Show More Info on Profile' is working again.

Also, after I downgraded to 1.7.2 (I cleared data, uninstalled, then installed using the 1.7.2 apk downloaded directly from here), I manually placed the 84609.json bindings in /data/data/xyz.rodit.snapmod/files/, but Snapchat kept freezing when opening. It would either get stuck on the yellow splash screen with the white ghost logo, or it would open to the camera window and be completely frozen. The camera circle would be half grayed out too, it was strange. Anyway, I ended up deleting the bindings file from there and placing it into /Android/data/xyz.rodit.snapmod/files/ instead, and now everything works fine (except the video freezing issue that persists). Do you know why manually placing the file in the root directory would be causing an issue? One thing I will note is that using 'Root Browser', when I go to the root directory and paste the file there, it will say "Root Browser was granted Superuser rights" and the file will be 0 KB. But if I exit and then return to the folder, it will now show as being 24 KB, which is normal. When pasting the file the same way into the internal storage directory, there's no need for any granting of Superuser rights, and the file gets pasted and shows as 24 KB straight away. I'm thinking something there is causing an issue. Maybe if I used a different Root explorer. Any recommendations?

Alright before posting this message I decided to downgrade Snapchat to 11.79.0.34 on my unrooted/unmodded receiving device, and yet I still experience >10s split videos freezing. It always happens after the first clip. The first clip plays fine, then the 2nd freezes midway, and the rest follow suit. Do you think perhaps there could be an AB testing issue to all of this? Maybe it just started with the account being used on my receiving device, but I hadn't noticed until updating to the latest Snapchat and SnapMod, so I incorrectly attributed it to the updates when in fact it could be AB testing related, which is why rolling back to an older Snapchat and SnapMod on both devices still results in the same issues. It's not version related. It's AB testing related. What do you think?

**Comment by rodit** [comment]
Videos freezing is usually fixed by restarting your phone. I doubt Snap are ab testing freezing videos lol. I'm not sure why placing the mappings in manually wouldn't work - you'd have to check xposed logs.

**Comment by Magna-Graecia** [comment]
Lmao no I don't mean that they're AB testing freezing videos. I mean maybe they're AB testing something else that's affecting it. I wish it were as simple as restarting my device hahaha. I had to restart a million times while downgrading, so ya it's def not that. Remember that I'm not talking about videos freezing on my device. I'm talking about them freezing on the recipient's end (the subject of this post).

**Comment by rodit** [comment]
I know haha - I implicitly meant I doubt Snap would allow such a bug through their testing. I'm assuming you have already tried other videos/recording new videos with Snap's camera >10s and then sending them to the recipient device?

**Comment by Magna-Graecia** [comment]
Yep, I sent 4 different videos. One was 1:05 long, another was 57 seconds long, another was 35 seconds long, and then the last was recorded from my camera and sent straight. That one was 60 seconds long (I locked the recording and let it go the full way). They all experienced the same issue. I just updated Snapchat and SnapMod to the latest version and tried all 4 again and still got the same results unfortunately. Not sure what's causing it. What I noticed is that the freezing happens on the 2nd clip EXACTLY as the timer bubble is halfway through (so at 5 seconds basically). The third clip will freeze at slightly less (so maybe at the 4 second point?). The 4th clip will freeze at an even lesser point (3 seconds?) and so on. It's crazy weird. I know that Snap wouldn't allow this type of bug through, but what I'm saying is maybe they're AB testing something that's causing the issue for some of us users using SnapMod. Same as how something that they were testing was causing issues with some features of stealth mode for some of us. That's not a bug that they're going to detect. As far as they're concerned, there is no stealth mode haha. Stealth mode only exists for us users on SnapMod. It would be great if @snaphappy1 could test since I know they're using the same setup as me, but not sure if they're still active.

One thing I can say is HOLY CRAP, the new compression/transcoding disabling is 1000% working. You're a freakin genius lmao. I set video resolution to the highest setting, but didn't touch the others (not sure what a proper number to set them to would be), and tried with a video that was always problematic for me, and the difference is night and day. I noticed a couple things though. One is that compression still happens when posting to story. I tried with that same video with both 'Force Encoding' turned on and off, and they got compressed both ways when posting to story, but less-so when 'Force Encoding' was enabled. There's definitely a difference. Compression especially happens when adding captions/filters to a video. And that brings me to the other thing I noticed. When encoding is enabled, I can't add captions to some photos. If disabled, it works fine. It doesn't happen with all photos, but it does happen with some. I wonder why. If I set video resolution to the highest setting and send a video with a source resolution that's less than that, does it just fall back on whatever the source resolution is? Should I try setting bitrate and FPS to a particular number in conjunction with the highest resolution setting to get even better quality? Can you recommend some numbers to me?

Also, for the record, 'Disable Snap Splitting' is still working fine for me, since I see that some others are still having issues with it. When the option is enabled, videos get sent in 1 snap and no freezing happens, so that's good at least. It's crazy how we're all experiencing different things.

**Comment by Magna-Graecia** [comment]
Man I don't know what you did, but videos and images are looking SOOOO good now lmao. I changed bitrate to 100,000,000 bps because why not :laughing:. Quality is everything! I haven't touched FPS though. I guess it might just make some videos smoother? Even when posting to story WITH a caption/location filter, the degradation is very minimal. It can be hard to notice it. Some videos are more noticeable than others. I've been testing with a lot of images and videos, and this is a game changer. Please don't change anything and break it :laughing:
